src/components/MapLille.jsx

```diff
@@ -37,26 +48,61 @@ class MapLille extends React.Component {
       ? this.props.coords.latitude
       : DEFAULT_LATITUDE;
 
+    function iconSelect(index) {
+      let fillingRate =
+        stations[index].fields.nbplacesdispo /
+        (stations[index].fields.nbplacesdispo +
+          stations[index].fields.nbvelosdispo);
+      if (fillingRate === 0) {
+        return icon0;
+      } else if (fillingRate < 0.33) {
+        return icon1;
+      } else if (fillingRate < 0.66) {
+        return icon2;
+      } else if (fillingRate < 0.1) {
+        return icon3;
+      } else if (fillingRate === 1) {
+        return icon4;
+      } else {
+        return iconPb;
+      }
+    }
+
     return (
       <div>
-        <Map center={[latitude, longitude]} zoom={14}>
+        <Map center={[latitude, longitude]} zoom={14} minZoom={11}>
           <TileLayer
             url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
             attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
           />
 
-          {stations.map((station) => (
+          {stations.map((station, index) => (
             <Marker
-              key={station.fields.nom}
+              key={index}
               position={[
                 station.geometry.coordinates[1],
                 station.geometry.coordinates[0],
               ]}
+              icon={L.icon({
+                iconUrl: iconSelect(index),
+                iconRetinaUrl: iconSelect(index),
+                shadowUrl: iconShadow,
+                iconSize: [38, 95],
+                iconAnchor: [22, 94],
+                shadowAnchor: [12, 42],
+                popupAnchor: [0, -85],
```

Comment:
I think there is a problem about markers size on the map, can you check this ?

Reply:
Il faudra qu'on se cale ensemble sur les tailles, de mon côté il charge les markers en version "retina" (taille différente)